### PR TITLE
AP_HAL_CHIBIOS: move serial params from defaults to hwdef

### PR DESCRIPTION
**Pull request by Hwurzburg**
Moving the serial param out of defaults and into hwdefs....fixed some duplications due to inheritance or plain duplication also

this is wip becuase I need to change SERIAL_MANAGER to use the new DEFAULT_X pattern for defaults....a separate PR...

### CODE REVIEW

**Comment by magicrub** [comment]
I guess there's a define missing?
```
../../libraries/AP_SerialManager/AP_SerialManager.cpp:68:22: error: 'HAL_SERIAL6_BAUD' was not declared in this scope; did you mean 'HAL_SERIAL5_BAUD'?
   68 | #define SERIAL6_BAUD HAL_SERIAL6_BAUD
```

**Comment by Hwurzburg** [comment]
the size CI does not include these boards...the size will grow for each character for each of these, since the text is stored in ROMFS for the hwdef now....but getting rid of the defaults.parm can eventually save 1K on 1MB boards and 8K on 2MB boards since we can get rid of the APJ reserved space for this...that is why we are moving things back into hwdef.dat

**Comment by Hwurzburg** [comment]
@andypiper you might want to look this over for your boards....many boards inherited the serial protocols and adding them to their derivative hwdef was not needed.....

File: libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef.dat [hunk 1]
```diff
@@ -83,9 +83,10 @@ define BOARD_RSSI_ANA_PIN 8
 
 PC2 LED0 OUTPUT LOW
 
-# USART1
+# USART1 DJI Goggles
 PA9  USART1_TX USART1
 PA10 USART1_RX USART1
+define HAL_SERIAL1_PROTOCOL 33
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/KakuteH7/hwdef.dat:90]: Missing SERIAL7 protocol 16

**Review by andyp1per — CHANGES_REQUESTED** [review]
Just one problem I found

**Review by andyp1per — APPROVED** [review]

**Comment by Hwurzburg** [comment]
thanks...but still wip...have to change the hwdef names...after my SerialManage normalization PR gets in...that will allow move the serial options and make the bauds actually work...

**Comment by peterbarker** [comment]
That's a lot of conflicts, @Hwurzburg !

**Comment by Hwurzburg** [comment]
Yep....may eventually get around to working on this again...

**Pull request closed without merging**